### PR TITLE
Fix event binding bug in swipe-composer

### PR DESCRIPTION
**Pull request by kevinluosl**
Fix in swipe that should use this._element instead document to bind event to target element

### CODE REVIEW

**Comment by rayshan** [comment]
Just curious, is there a difference b/t `this._element` and `this.element`?

**Comment by kevinluosl** [comment]
From the codes there is no different. When I was doing this modify, I saw other composer using this._element so I followed.

**Comment by rayshan** [comment]
My experience is that they work the same as well. I'll ask around.

**Comment by rayshan** [comment]
I discussed w/ @thibaultzanini, summary:

- Don't get or set `_privateProperty` directly due to possible custom getter / setters
- Can call methods on `_privateProperty` directly for better performance, e.g. `this._element.addEventListener`

I'll add this to docs at some point.

File: composer/swipe-composer.js [hunk 1]
```diff
@@ -193,7 +199,7 @@ exports.SwipeComposer = Composer.specialize( /** @lends SwipeComposer# */ {
                 return;
             }
 
-            document.removeEventListener("touchmove", this, true);
+            this._removeEventListeners();
```
Comment (rayshan) [inline composer/swipe-composer.js]: Why do this right here, not the end of `captureTouchend` block?

**Comment by rayshan** [comment]
Hi @kevinluosl, can we have some background on why events should be on `this._element` instead `document`? If we're using `this._element`, should we still use capture instead of the default bubble?

File: composer/swipe-composer.js [hunk 2]
```diff
@@ -78,9 +78,9 @@ exports.SwipeComposer = Composer.specialize( /** @lends SwipeComposer# */ {
             this._startX = touch.clientX;
             this._startY = touch.clientY;
             this._startTimestamp = event.timeStamp;
-            document.addEventListener("touchmove", this, true);
-            document.addEventListener("touchend", this, true);
-            document.addEventListener("touchcancel", this, true);
+            this._element.addEventListener("touchmove", this, true);
+            this._element.addEventListener("touchend", this, true);
+            this._element.addEventListener("touchcancel", this, true);
```
Comment (thibaultzanini) [inline composer/swipe-composer.js]: agreed for listening on the element swipe composer for unload and load but not for touchmove, touchend, and touchcancel

File: composer/swipe-composer.js (hunk 1 above)
Comment (thibaultzanini) [inline composer/swipe-composer.js]: I'm not sure If it's a big deal or not, but yeah in order to keep the same logic in our framework (we chosen that way for the translate composer for exemple)

**Comment by thibaultzanini** [comment]
@kevinluosl can you update that PR please.

**Pull request closed without merging**